02_configure_host.sh

```diff
@@ -245,7 +245,16 @@ for IMAGE_VAR in $(env | grep "_LOCAL_IMAGE=" | grep -o "^[^=]*") ; do
 
   # Support building ironic-image from source
   if [[ "${IMAGE}" =~ "ironic" ]] && [[ ${IRONIC_FROM_SOURCE:-} == "true" ]]; then
-    sudo "${CONTAINER_RUNTIME}" build --build-arg INSTALL_TYPE=source -t "${!IMAGE_VAR}:latest" -t "${!IMAGE_VAR}:${IMAGE_GIT_HASH}_${IMAGE_DATE}" . -f ./Dockerfile
+    # NOTE(rpittau): to customize the source origin we need to copy the source code we
+    # want to use into the sources directory under the ironic-image repository.
+    for CODE_SOURCE_VAR in $(env | grep -E '^IRONIC_SOURCE=|^IRONIC_INSPECTOR_SOURCE=|^SUSHY_SOURCE=' | grep -o "^[^=]*"); do
+      CODE_SOURCE="${!CODE_SOURCE_VAR}"
+      SOURCE_DIR_DEST="${CODE_SOURCE##*/}"
+      cp -a "$CODE_SOURCE" "./sources/${SOURCE_DIR_DEST}"
+      CUSTOM_SOURCE_ARGS+="--build-arg ${CODE_SOURCE_VAR}=${SOURCE_DIR_DEST} "
+    done
+    #shellcheck disable=SC2086
+    sudo "${CONTAINER_RUNTIME}" build --build-arg INSTALL_TYPE=source ${CUSTOM_SOURCE_ARGS:-} -t "${!IMAGE_VAR}:latest" -t "${!IMAGE_VAR}:${IMAGE_GIT_HASH}_${IMAGE_DATE}" . -f ./Dockerfile
```

Comment:
It is unnecessary here to disable shellcheck. We can instead do it properly with a bash array. Something like this should do it:

```bash
declare -a CUSTOM_SOURCE_ARGS
for CODE_SOURCE_VAR in $(env | ...); do
    ...
    CUSTOM_SOURCE_ARGS+=("--build-arg ...")
done
sudo "${CONTAINER_RUNTIME}" build ... "${CUSTOM_SOURCE_ARGS[@]}"
```

Reply:
have you tested it?
I tried that but it didn't work as it's still read as a single string, maybe I did something wrong?

Reply:
Yes, we use it in the 03 script. Here we create the array: https://github.com/metal3-io/metal3-dev-env/blob/3c9dcb188c05a559383257ebdacfa3897fbfee14/03_launch_mgmt_cluster.sh#L17-L28
And here we use it: https://github.com/metal3-io/metal3-dev-env/blob/3c9dcb188c05a559383257ebdacfa3897fbfee14/03_launch_mgmt_cluster.sh#L72
It must be an array though, otherwise it is always treated as a single string when quoted.

Reply:
Never mind. I didn't consider the empty case! It would break if the array is empty, since you then end up with just `""`.